Zend/zend_object_handlers.c

```diff
@@ -1020,8 +1020,10 @@ ZEND_API zval *zend_std_get_property_ptr_ptr(zend_object *zobj, zend_string *nam
 						ZVAL_NULL(retval);
 						zend_error(E_WARNING, "Undefined property: %s::$%s", ZSTR_VAL(zobj->ce->name), ZSTR_VAL(name));
 					}
-				} else if (prop_info && UNEXPECTED(prop_info->flags & ZEND_ACC_READONLY)
-					&& !verify_readonly_initialization_access(prop_info, zobj->ce, name, "initialize")) {
+				} else if (prop_info && UNEXPECTED(prop_info->flags & ZEND_ACC_READONLY)) {
+					if (verify_readonly_initialization_access(prop_info, zobj->ce, name, "initialize")) {
+						zend_readonly_property_modification_error(prop_info);
+					}
```

Comment:
I don't think throwing an unconditional error is right here -- shouldn't this return null to fall back to read_property/write_property?

Reply:
Ah ok. I'm guessing `verify_readonly_initialization_access` is only in `zend_std_get_property_ptr_ptr` because passing `NULL` would call `read_property` before `write_property` which would cause an "Typed property must not be accessed before initialized" error message, instead of the scope error message?

That this will also cause something like `$c->a[] = 1;` to fail with a "Typed property must not be accessed before initialized" message, which is technically right since we're falling back to `read_property` but might be unintuitive.

Reply:
@nikic done.